pkg/controller/datavolume-controller.go

```diff
@@ -2541,14 +2572,28 @@ func getDefaultVolumeMode(c client.Client, storageClass *storagev1.StorageClass)
 	}
 	if len(storageProfile.Status.ClaimPropertySets) > 0 {
 		volumeMode := storageProfile.Status.ClaimPropertySets[0].VolumeMode
+		if len(pvcAccessModes) == 0 {
+			return volumeMode, nil
+		}
+		// check for volume mode matching with given pvc access modes
+		for _, cps := range storageProfile.Status.ClaimPropertySets {
+			for _, accessMode := range cps.AccessModes {
+				for _, pvcAccessMode := range pvcAccessModes {
+					if accessMode == pvcAccessMode {
+						return cps.VolumeMode, nil
+					}
+				}
+			}
+		}
+		// if not found return default volume mode for the storage class
```

Comment:
If accessModes are defined but not found - it might mean that a user defined accessMode that is not supported, but I think our current logic is correct. 

The effect would be the pvc could not be bound, and the user can check the PVC events and learn which accessModes are suported on the selected storage.

Reply:
@brybacki Not sure I understood. You mean that if len(pvcAccessModes) == 0 or we didnt find in the loop the access mode then we should return nil? or if len(pvcAccessModes) == 0 return the storageProfile.Status.ClaimPropertySets[0].VolumeMode and in case we didnt find in the loop we should return error?

Reply:
I mean, we did not find in the loop. But maybe all this options are similar. I need to think about this again.